R/gh_request.R

```diff
@@ -108,7 +108,7 @@ gh_set_url <- function(x) {
     x$url <- URLencode(x$endpoint)
     x$api_url <- get_baseurl(x$url)
   } else {
-    x$api_url <- x$api_url %||% default_api_url()
+    x$api_url <- get_apiurl(x$api_url %||% default_api_url())
```

Comment:
Makes it so we can (quietly) accept "https://github.com" or "https://github.acme.com" instead of insisting on "https://api.github.com" or "https://github.acme.com/api/v3".

Reply:
This makes sense as a global attitude, since GitHub.com PATs are already keyed with "https://github.com" in the Git credential store, as opposed to "https://api.github.com". I think we should extend the same policy to GHE PATs and, since we do token lookup based on `.api_url`, we eventually arrive at this.